iati/core/tests/test_rulesets.py

```diff
@@ -440,452 +558,670 @@ def valid_dataset(self):
+        return {'paths': ['element13', 'element14/@attribute']}
+
+    @pytest.fixture
+    def invalid_dataset(self):
+        """Invalid dataset for this Rule."""
+        return iati.core.tests.utilities.DATASET_FOR_NOMORETHANONE_RULE_INVALID
+
+    @pytest.fixture
+    def valid_dataset(self):
+        """Return valid dataset for this Rule."""
+        return iati.core.tests.utilities.DATASET_FOR_NOMORETHANONE_RULE_VALID
+
+
+class TestRuleRegexMatches(RuleSubclassTestBase):
+    """A container for tests relating to RuleRegexMatches."""
+
+    all_valid_cases = [
+        {'regex': r'\btest\b', 'paths': ['element1']},  # single path with regex
+        {'regex': r'\btest\b', 'paths': ['element5/@attribute']},
+        {'regex': r'\btest\b', 'paths': ['element2', 'element3']},  # multiple paths with regex
+        {'regex': r'\btest\b', 'paths': ['element6/@attribute', 'element7/@attribute']},
+        {'regex': r'\btest\b', 'paths': ['element4', 'element4']},  # duplicate paths with regex
+        {'regex': r'\btest\b', 'paths': ['element8/@attribute', 'element8/@attribute']},
+        {'regex': r'\btest\b', 'paths': ['element11']}  # duplicate element
+    ]
+
+    uninstantiating_cases = [
+        {'regex': r'some regex', 'paths': []},  # empty path array
+        {'regex': r'some regex', 'paths': ['']},  # paths is an empty string
+        {'regex': r'some regex', 'paths': 'path_1'},  # non-array `paths`
+        {'regex': r'some regex', 'paths': [3]},  # non-string value in path array
+        {'regex': r'some regex', 'paths': ['path_1', 3]},  # mixed string and non-string value in path array
+        {'regex': r'some regex'},  # missing required attribute - `paths`
+        {'paths': ['path_1', 'path_2']},  # missing required attribute - `regex`
+        {'regex': r'[', 'paths': ['path_1']},  # provided string not a valid regex
+        {'regex': 3, 'paths': ['path_1']},  # provided regex not a string
+        {},  # empty dictionary
+        {'regex': r'some regex', 'paths': {'path_1'}},  # dictionary paths
+        {'regex': [r'some regex'], 'paths': 'path_1'},  # list regex
+        {'regex': r'', 'paths': ['path_1']}  # regex is an empty string
+    ]
+
+    invalidating_cases = [
+        {'regex': r'\btest\b', 'paths': ['element1']},  # single path with regex
+        {'regex': r'\btest\b', 'paths': ['element5/@attribute']},
+        {'regex': r'\btest\b', 'paths': ['element2', 'element3']},  # multiple paths with regex
+        {'regex': r'\btest\b', 'paths': ['element6/@attribute', 'element7/@attribute']},
+        {'regex': r'\btest\b', 'paths': ['element4', 'element4']},  # duplicate paths with regex
+        {'regex': r'\btest\b', 'paths': ['element8/@attribute', 'element8/@attribute']},
+        {'regex': r'\btest\b', 'paths': ['element11']}  # duplicate element
+    ]
+
+    nest_case = [{'regex': r'\btest\b', 'paths': ['./element9', './element10/@attribute']}]
```

Comment:
Should be plural since it's an array

Reply:
no, because it's only one case. It's in an array so it can be used by the appropriate fixtures.

Reply:
More could be added since it's an array. Also, `TestRuleSum` has a plural name. It should be consistent.

Reply:
TestRuleSum has multiple cases.

Reply:
Consistency issues should be part of a general test data refactor effort that is not within the scope of this branch.